arm/Microsoft.Network/virtualNetworkGateways/deploy.bicep

```diff
@@ -189,8 +192,8 @@ var gatewaySubnetId = '${vNetResourceId}/subnets/GatewaySubnet'
 var activeActive_var = virtualNetworkGatewayType == 'ExpressRoute' ? false : activeActive
 
 // Public IP variables
-var gatewayPipName1 = length(gatewayPipName) == 0 ? '${name}-pip1' : gatewayPipName[0]
-var gatewayPipName2 = activeActive_var ? (length(gatewayPipName) == 1 ? '${name}-pip2' : gatewayPipName[1]) : ''
+var gatewayPipName1 = empty(gatewayPipName) ? '${name}-pip1' : gatewayPipName
+var gatewayPipName2 = activeActive_var ? (empty(activeGatewayPipName) ? '${name}-pip2' : activeGatewayPipName) : ''
```

Comment:
I'd suggest to move both PIP names to the default values of the corresponding parameter.

E.g. `param activeGatewayPipName = '${name}-pip2'`

And then use the condition `activeActive_var` at the place where the parameter is used so that the way it works doesn't change.

Reply:
Done but keeping the variables used in the for loops of the pip creation.

Reply:
@MrMCake